disnake/ui/item.py

```diff
@@ -109,14 +124,14 @@ def __init__(self):
         # only called upon edit and we're mainly interested during initial creation time.
         self._provided_custom_id: bool = False
 
-    def refresh_component(self, component: Component) -> None:
+    def refresh_component(self, component: NestedComponent) -> None:
```

Comment:
Pretty sure `NestedComponent` here is correct, but it causes typechecking issues in other places, which would probably have to be updated as well:
```
disnake\ui\view.py:424:41 - error: Argument of type "Component | Button" cannot be assigned to parameter "component" of type "NestedComponent" in function "refresh_component"
    Type "Component | Button" cannot be assigned to type "NestedComponent"
      Type "Component" cannot be assigned to type "NestedComponent"
        "Component" is incompatible with "Button"
        "Component" is incompatible with "SelectMenu"
        "Component" is incompatible with "InputText" (reportGeneralTypeIssues)
```